scvi/model/base/_base_model.py

```diff
@@ -64,6 +69,8 @@ class BaseModelClass(metaclass=BaseModelMetaClass):
     """Abstract class for scvi-tools models."""
 
     def __init__(self, adata: Optional[AnnOrMuData] = None):
+        latent_adata = adata is not None and _get_latent_adata_type(adata) is not None
+        _raise_if_missing_latent_mode_support(type(self).__name__, latent_adata)
```

Comment:
why is this necessary?

Reply:
let's think if this works for developers in other codebases

Reply:
good call